daemon/task.go

```diff
@@ -103,44 +112,102 @@ func FormatTime(t time.Time) string {
 
 // Map the task onto a map[string]interface{}, using the given route for the Location()
 func (t *Task) Map(route *mux.Route) map[string]interface{} {
+	t.Lock()
+	defer t.Unlock()
+
 	return map[string]interface{}{
 		"resource":   t.Location(route),
 		"status":     t.State(),
-		"created_at": FormatTime(t.CreatedAt()),
-		"updated_at": FormatTime(t.UpdatedAt()),
+		"created_at": FormatTime(t.t0),
+		"updated_at": FormatTime(t.tf),
 		"may_cancel": false,
-		"output":     t.Output(),
+		"output":     t.output,
+	}
+}
+
+var (
+	// ErrNoReceiver is returned from Send when the task isn't listening (yet)
+	ErrNoReceiver = errors.New("task is not receiving messages at this time")
+	// ErrNotRunning is returned from Send when the task isn't Running
+	ErrNotRunning = errors.New("task is not running")
+)
+
+// Send a value into the task
+func (t *Task) Send(v interface{}) error {
+	t.Lock()
+	defer t.Unlock()
+
+	if t.State() != TaskRunning {
+		return ErrNotRunning
+	}
+
+	if t.chin == nil {
+		return ErrNoReceiver
 	}
+
+	t.output = nil
+	t.chin <- v
```

Comment:
There's another race here, and avoiding it is the main reason why the Dying and Dead tomb methods exist: the task may be dead right after the State test above, in which case this will never move on. Something like this should work:

    select {
    case <-t.tomb.Dying():
            return ErrNotRunning
    case t.chin <- v:
    }

This also removes the need for checking the State above altogether, since it's racy anyway.